### PR TITLE
Onboarding AKS Fleet Manager

### PR DESCRIPTION
**Pull request by Ealianis**
## Overview
Adding support for new Azure Kubernetes Fleet Manager. As instructed [here on step 2](https://applensdiagnostics.azurewebsites.net/?p=76).

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Checklist <span>&#9989;</span>
- [x] I have looked over the diffs.
- [x] I have run the linter to catch any errors.
- [x] There are no errors from my changes on this branch.

### CODE REVIEW

File: AngularApp/projects/applens/src/assets/enabledResourceTypes.json [hunk 1]
```diff
@@ -92,6 +92,18 @@
             "searchSuffix": "AZURE KUBERNETES CLUSTER",
             "displayName": "Kubernetes Service (AKS)"
         },
+        {
+            "service": "ResourceService",
+            "resourceType": "Microsoft.ContainerService/fleets",
+            "templateFileName": "KubernetesFleetManager",
+            "imgSrc": "assets/img/Azure-Kubernetes-FleetManager-Logo.png",
+            "versionPrefix": "",
+            "azureCommImpactedServicesList": "",
```
Comment (Ealianis) [inline AngularApp/projects/applens/src/assets/enabledResourceTypes.json:101]: Need to determine this list value.

**Review by jim-minter — APPROVED** [review]
LGTM

File: ApplensBackend/Templates/Detector_KubernetesFleetManager.csx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    return
+    $@"
+        let startTime = datetime({cxt.StartTime});
+        let endTime = datetime({cxt.EndTime});
+        cluster('Aks').database('AKSprod').<YOUR_TABLE_NAME>
+        | where Timestamp >= startTime and Timestamp <= endTime
+        | <YOUR_QUERY>";
+}
+
+[AzureKubernetesFleetManagerFilter]
```
Comment (nmallick1) [inline ApplensBackend/Templates/Detector_KubernetesFleetManager.csx:12]: `[ArmResourceFilter(provider: "Microsoft.ContainerService", resourceTypeName: "fleets")]`

File: ApplensBackend/Templates/Detector_KubernetesFleetManager.csx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+[Definition(Id = "<YOUR_DETECTOR_ID>", Name = "", Author = "<YOUR_ALIAS>", Description = "")]
+public async static Task<Response> Run(DataProviders dp, OperationContext<AzureKubernetesFleetManager> cxt, Response res)
```
Comment (nmallick1) [inline ApplensBackend/Templates/Detector_KubernetesFleetManager.csx:14]:
```suggestion
public async static Task<Response> Run(DataProviders dp, OperationContext<ArmResource> cxt, Response res)
```

File: ApplensBackend/Templates/Gist_KubernetesFleetManager.csx [hunk 4]
```diff
@@ -0,0 +1,6 @@
+
+[AzureKubernetesFleetManagerFilter]
```
Comment (nmallick1) [inline ApplensBackend/Templates/Gist_KubernetesFleetManager.csx:2]:
```suggestion
[ArmResourceFilter(provider: "Microsoft.ContainerService", resourceTypeName: "fleets")]
```

File: AngularApp/projects/applens/src/assets/enabledResourceTypes.json (hunk 1 above)
Comment (Ealianis) [reply AngularApp/projects/applens/src/assets/enabledResourceTypes.json]: Verified this is just  a display string, updated to "Azure Kubernetes Fleet Manager".

**Review by nmallick1 — APPROVED** [review]

File: ApplensBackend/Templates/Detector_KubernetesFleetManager.csx [hunk 5]
```diff
@@ -9,9 +9,9 @@ private static string GetQuery(OperationContext<AzureKubernetesFleetManager> cxt
         | <YOUR_QUERY>";
 }
```
[the remark this answers is not in the document]
Comment (Ealianis) [reply ApplensBackend/Templates/Detector_KubernetesFleetManager.csx:10]: Saw this after merge was complete. Will update in future PR should we update these base templates soon.

**Pull request merged**